projects/observability/src/shared/dashboard/widgets/topology/edge/curved/entity-edge-curve-renderer.service.ts

```diff
@@ -90,13 +94,27 @@ export class EntityEdgeCurveRendererService implements TopologyEdgeRenderDelegat
     domRenderer: Renderer2
   ): void {
     const selection = this.d3Utils.select(element, domRenderer);
-    const metricSpecifications = state.dataSpecifiers?.map(specifier => specifier.value);
+
+    const primaryMetric = this.topologyDataSourceModelPropertiesService.getPrimaryEdgeMetric();
+    const primaryMetricAggregation = getTopologyMetric(edge.data, primaryMetric?.specification);
+    const primaryMetricCategory = getPrimaryEdgeMetricCategory(
+      primaryMetricAggregation?.value,
+      primaryMetric?.categories
+    );
+    const secondaryMetric = this.topologyDataSourceModelPropertiesService.geSecondaryEdgeMetric();
+    const secondaryMetricAggregation = getTopologyMetric(edge.data, secondaryMetric?.specification);
```

Comment:
If all of these are static properties, then we can assign them in constructor. We can just call a method that can set all of these as instance properties. No need to call this method every render cycle